### PR TITLE
Implement CreateCopyJob

### PR DESCRIPTION
**Pull request by jskeet**
(2 separate commits, probably simplest to review separately. The first commit is just refactoring.)

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQuerySnippetFixture.cs [hunk 1]
```diff
@@ -84,6 +84,10 @@ private string CreateGameDataset()
 
         internal string GenerateDatasetId() => DatasetPrefix + Guid.NewGuid().ToString().Replace('-', '_');
 
+        internal string GenerateTableId() => Guid.NewGuid().ToString().Replace("-", "_");
+
+        internal string GenerateStorageBucketName() => Guid.NewGuid().ToString().Replace("-", "_");
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQuerySnippetFixture.cs]: This doesn't conform to how this code used to name storage buckets. Is this intentional?

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryClientSnippets.cs [hunk 2]
```diff
@@ -1127,11 +1162,47 @@ public void ParameterizedQuery_PositionalParameters()
...
 
         [Fact]
+        public async Task CreateCopyJobAsync()
+        {
+            string projectId = _fixture.ProjectId;
+            string datasetId = _fixture.GameDatasetId;
+            string sourceTableId = _fixture.HistoryTableId;
+            string destinationTableId = _fixture.GenerateTableId();
+
+            // Snippet: CreateExtractJob(string, string, string, string, *)
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryClientSnippets.cs]: Missing Async in snippet name

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClient.JobCrud.cs [hunk 3]
```diff
@@ -238,6 +238,39 @@ public virtual BigQueryJob CreateExtractJob(TableReference tableReference, IEnum
...
+        /// <param name="source">The source table to copy. Must not be null.</param>
+        /// <param name="destination">The destination to copy to. Must not be null.</param>
+        /// <param name="options">The options for the operation. May be null, in which case defaults will be supplied.</param>
+        /// <returns>The job created for the copy operation.</returns>
+        public virtual BigQueryJob CreateCopyJob(TableReference source, TableReference destination, CreateCopyJobOptions options = null)
+            => CreateCopyJob(new[] { GaxPreconditions.CheckNotNull(source, nameof(source)) }, destination, options);
+
+        /// <summary>
+        /// Creates a job to copy data from at least one table to another.
+        /// This method creates a single-element array and delegates to <see cref="CreateCopyJob(IEnumerable{TableReference}, TableReference, CreateCopyJobOptions)"/>.
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClient.JobCrud.cs]: This line should be in the above method.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQuerySnippetFixture.cs (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQuerySnippetFixture.cs]: No, copy-paste fail.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryClientSnippets.cs (hunk 2 above)
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryClientSnippets.cs]: Missing the right snippet name, too...

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClient.JobCrud.cs (hunk 3 above)
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClient.JobCrud.cs]: Aargh... I deliberately removed it from one, but apparently the wrong one. I'm rubbish.

**Comment by jskeet** [comment]
Fixed problems - running snippet tests, will merge when green and all tests pass.

**Pull request merged**